### ISSUE TITLE
Add connection pool configuration

### ISSUE THREAD

**Issue by RoyY1234** [swiftcarrot/queryx#92]
By my test, add connection pool configuration could resolve some db socket connection error.
Currently, I modify manually `queryx.go` like this:
```
func NewClientWithEnv(env string) (*QXClient, error) {
	config := queryx.NewConfig(env)
	if config == nil {
		return nil, fmt.Errorf("client config is missing for %s", env)
	}
	db, err := sql.Open("postgres", config.URL)
	if err != nil {
		return nil, err
	}

	db.SetMaxOpenConns(300)
	db.SetMaxIdleConns(100)
	db.SetConnMaxLifetime(3 * time.Second)

        ...
```

Do you consider make this configurable like database connection setting?

**Comment by wangzuo** [comment]
Instead of implementing these connection options in schema, what about

```golang
c, err := db.NewClient()

c.SetMaxOpenConns(300)
c.SetMaxIdleConns(100)
c.SetConnMaxLifetime(3 * time.Second)
```

**Comment by RoyY1234** [comment]
This solution looks much reasonable! But you need to add extra adaptor functions.
Maybe it's safe to expose db to developer, like
```
type QXClient struct {
	DB     *sql.DB     // make DB public
	config *queryx.Config
	logger queryx.Logger
	*queryx.Adapter
	*queryx.Schema
}
```
then developer could manipulate db by themself.
```
c, err := db.NewClient()

c.DB.SetMaxOpenConns(300)
c.DB.SetMaxIdleConns(100)
c.DB.SetConnMaxLifetime(3 * time.Second)
```

**Comment by wangzuo** [comment]
Thanks @RoyY1234 , I will roll it out soon.

**Comment by wangzuo** [comment]
This has been released in `v0.2.13`.
